account_invoice_triple_discount/migrations/16.0.2.0.0/post-migrate.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.logging()
+def compute_discount(env):
+    move_lines_to_compute = env["account.move.line"].search(
+        [
+            "|",
+            "|",
+            ("discount1", "!=", 0),
+            ("discount2", "!=", 0),
+            ("discount3", "!=", 0),
+        ]
+    )
+    for line in move_lines_to_compute:
+        discount = line._get_aggregated_multiple_discounts(
+            [line[x] for x in ["discount1", "discount2", "discount3"]]
+        )
+        rounded_discount = line._fields["discount"].convert_to_column(discount, line)
+        openupgrade.logged_query(
+            env.cr,
+            """
+            UPDATE account_move_line
+            SET discount = %s
+            WHERE id = %s;
+        """,
+            tuple(
+                [
+                    rounded_discount,
+                    line.id,
+                ]
+            ),
+        )
+
```

Comment:
This migration script is unusable on a large database :sob: The value should be updated in one single SQL query.